### PR TITLE
Fix backtest duration calculation

### PR DESCRIPTION
**Pull request by xmatthias**
## Summary
The calculation of trade duration is wrong if a trade takes longer than 1 day, as seconds does not return the full second count.

Solve the issue: #1038

## quick changelog:
* fix duration calculation for longer trades
* readable output (5000 s does not tell me how long that is).

## old output:
```
================================================= BACKTESTING REPORT ==================================================
| pair      |   buy count |   avg profit % |   cum profit % |   total profit ETH |   avg duration |   profit |   loss |
|:----------|------------:|---------------:|---------------:|-------------------:|---------------:|---------:|-------:|
| LTC/ETH   |          50 |           0.67 |          33.42 |         0.03345814 |          130.0 |       49 |      1 |
| IOTA/ETH  |          82 |           0.90 |          73.77 |         0.07384440 |          133.9 |       80 |      2 |
| XMR/ETH   |          65 |           0.80 |          52.12 |         0.05216980 |          121.0 |       63 |      2 |
| XRP/ETH   |          52 |           0.85 |          44.09 |         0.04413170 |          223.8 |       52 |      0 |
| NCASH/ETH |          67 |           0.57 |          37.95 |         0.03798353 |           82.2 |       62 |      5 |
| MTH/ETH   |          76 |           0.80 |          61.04 |         0.06109623 |           94.0 |       72 |      4 |
| AST/ETH   |          77 |           1.16 |          89.49 |         0.08957901 |          110.4 |       76 |      1 |
| TNT/ETH   |          88 |           1.02 |          89.55 |         0.08963916 |          120.7 |       80 |      8 |
| ETC/ETH   |          73 |           0.66 |          48.06 |         0.04810711 |          175.1 |       68 |      5 |
| TOTAL     |         630 |           0.84 |         529.48 |         0.53000908 |          129.4 |      602 |     28 |
```

## new output:
```
================================================= BACKTESTING REPORT ==================================================
| pair      |   buy count |   avg profit % |   cum profit % |   total profit ETH | avg duration   |   profit |   loss |
|:----------|------------:|---------------:|---------------:|-------------------:|:---------------|---------:|-------:|
| LTC/ETH   |          70 |           0.74 |          52.10 |         0.05215068 | 3:54:00        |       69 |      1 |
| IOTA/ETH  |         100 |           0.90 |          89.66 |         0.08974745 | 3:02:00        |       98 |      2 |
| XMR/ETH   |          65 |           0.80 |          52.12 |         0.05216980 | 3:30:00        |       63 |      2 |
| XRP/ETH   |          52 |           0.85 |          44.09 |         0.04413170 | 6:58:00        |       52 |      0 |
| NCASH/ETH |          89 |           0.94 |          83.49 |         0.08357569 | 1:42:00        |       83 |      6 |
| MTH/ETH   |         107 |           0.88 |          94.21 |         0.09429964 | 1:59:00        |      101 |      6 |
| AST/ETH   |         120 |           1.23 |         147.24 |         0.14738299 | 1:53:00        |      117 |      3 |
| TNT/ETH   |          88 |           1.02 |          89.55 |         0.08963916 | 3:06:00        |       80 |      8 |
| ETC/ETH   |         100 |           0.73 |          72.75 |         0.07282143 | 4:13:00        |       95 |      5 |
| TOTAL     |         791 |           0.92 |         725.19 |         0.72591854 | 3:06:00        |      758 |     33 |


```

### CODE REVIEW

**Comment by creslinux** [comment]
This fixes the problem "technically" usability/usefulness may have gone out the window
The number created here is for a human to read, 8040 minutes is a bit savant , i needed to grab a calculator to know what i was being told. 

The open and close timestamps are human readable, but hidden from the report. 
We should provide this output in human readable format. 

```
2018-07-18 09:51:31,399 - freqtrade.optimize.backtesting - INFO - 
================================================= BACKTEST REPORT ==================================================
| pair    |   buy count |   avg profit % |   cum profit % |   total profit BTC |   avg duration |   profit |   loss |
|:--------|------------:|---------------:|---------------:|-------------------:|---------------:|---------:|-------:|
| ETH/BTC |           1 |          11.32 |          11.32 |         0.00170014 |         8040.0 |        1 |      0 |
| TOTAL   |           1 |          11.32 |          11.32 |         0.00170014 |         8040.0 |        1 |      0 |
```

Backslap used more readable D/H/M/S format:
```
2018-07-18 09:47:09,251 - freqtrade.optimize.backtesting - INFO - 
====================================================== BackSLAP REPORT =======================================================
| pair    |   buy count |   avg profit % |   cum profit % |   total profit BTC | avg duration    |   profit |   loss |
|:--------|------------:|---------------:|---------------:|-------------------:|:----------------|---------:|-------:|
| ETH/BTC |           1 |          11.43 |          11.43 |         0.00171601 | 5 days 14:00:00 |        1 |      0 |
| TOTAL   |           1 |          11.43 |          11.43 |         0.00171601 | 5 days 14:00:00 |        1 |      0 |
```

**Comment by xmatthias** [comment]
happy to accept a PR for this (either to replace this PR - or against the branch I'm doing the PR from if you'd like to keep my (fairly small) changes.

**Comment by creslinux** [comment]
nice!

File: freqtrade/optimize/backtesting.py [hunk 1]
```diff
@@ -112,7 +113,8 @@ def _generate_text_table(self, data: Dict[str, Dict], results: DataFrame) -> str
             results.profit_percent.mean() * 100.0,
             results.profit_percent.sum() * 100.0,
             results.profit_abs.sum(),
-            results.trade_duration.mean(),
+            str(timedelta(
+                minutes=round(results.trade_duration.mean()))) if len(results) else 'nan',
```
[the remark this answers is not in the document]
Comment (xmatthias) [reply freqtrade/optimize/backtesting.py]: This is a dataframe ... therefore 
`The truth value of a DataFrame is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().` 

but i agree, will change to results.empty

**Pull request merged**